src/main/java/in/org/projecteka/hiu/dataflow/DataFlowRepository.java

```diff
@@ -59,6 +62,18 @@
             "join consent_artefact ca on dfr.consent_artefact_id = ca.consent_artefact_id " +
             "where ca.consent_request_id in (%s)";
 
+    private static final String FETCH_PATIENT_DATA_REQUEST_DETAILS = "SELECT " +
+            "pcr.hip_id, pcr.data_request_id::text, " +
+            "pcr.date_created AS patient_data_requested_at, " +
+            "pcr.date_modified AS consent_requested_at, " +
+            "dfr.date_created AS data_flow_requested_at, " +
+            "dfp.status AS data_part_status, pcr.consent_request_id::text, ca.consent_artefact_id " +
+            "FROM patient_consent_request pcr " +
+            "LEFT JOIN consent_artefact ca ON pcr.consent_request_id = ca.consent_request_id::uuid " +
+            "LEFT JOIN data_flow_request dfr ON dfr.consent_artefact_id = ca.consent_artefact_id " +
+            "LEFT JOIN data_flow_parts dfp ON dfr.transaction_id = dfp.transaction_id " +
+            "WHERE pcr.data_request_id IN (%s)";
```

Comment:
Seems it will take longer time to update, can it be cached?

Reply:
I think it can't be cached, because it needs to return the live status for data request ids.

Reply:
but you can have cache where you keep updating the status whenever you process the record you received from HIP right? with few it might look ok, but when things grow, it will become slower.

Reply:
Ok got your point, But the status API is not just working on the data flow part status, there are other things, If you want I can give you more context.